lnbits/static/js/base.js

```diff
@@ -179,14 +179,79 @@ window.LNbits = {
   },
   events: {
     onInvoicePaid(wallet, cb) {
-      ws = new WebSocket(`${websocketUrl}/${wallet.inkey}`)
-      ws.onmessage = ev => {
-        const data = JSON.parse(ev.data)
-        if (data.payment) {
-          cb(data)
+      const reconnectInterval = 5000 // 5 seconds
+      const maxRetries = 12 // ~60 seconds
+      const websocketUrlFull = `${websocketUrl}/${wallet.inkey}`
+
+      if (!g.activeWebsockets) g.activeWebsockets = {}
+      if (!g.disconnectedWallets) g.disconnectedWallets = new Set()
+
+      const updateConnectionWarning = () => {
+        g.connectionWarning = g.disconnectedWallets.size > 0
+      }
+
+      const connect = (retryCount = 0) => {
```

Comment:
this should be a method of the component, not a variable inside an event callback function